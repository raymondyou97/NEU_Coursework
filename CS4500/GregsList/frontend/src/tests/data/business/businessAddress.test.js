'use es6';

import React from 'react';
import BusinessAddress from '../../../components/business/Address';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('Initial snapshot renders', () => {
  const component = renderer.create(<BusinessAddress />);

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Correct number of default forms and default values', () => {
  const component = shallow(<BusinessAddress />);

  expect(component.find('.form-control').length).toEqual(4);
  expect(component.find('#businessStreetInput').props().value).toEqual(
    undefined
  );
  expect(component.find('#businessCityInput').props().value).toEqual(undefined);
  expect(component.find('#businessZipInput').props().value).toEqual(undefined);
});
