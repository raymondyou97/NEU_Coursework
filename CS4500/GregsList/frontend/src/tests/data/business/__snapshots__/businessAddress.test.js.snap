// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial snapshot renders 1`] = `
<div>
  <h5>
    Business Address (optional)
  </h5>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <label
        htmlFor="businessStreetInput"
      >
        Street
      </label>
      <input
        className="form-control"
        id="businessStreetInput"
        type="text"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <label
        htmlFor="businessCityInput"
      >
        City
      </label>
      <input
        className="form-control"
        id="businessCityInput"
        type="text"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <label
        for="businessStateInput"
      >
        State
      </label>
      <select
        className="form-control"
        id="businessStateInput"
      >
        &gt;
        <option
          value="AL"
        >
          Alabama
        </option>
        <option
          value="AK"
        >
          Alaska
        </option>
        <option
          value="AZ"
        >
          Arizona
        </option>
        <option
          value="AR"
        >
          Arkansas
        </option>
        <option
          value="CA"
        >
          California
        </option>
        <option
          value="CO"
        >
          Colorado
        </option>
        <option
          value="CT"
        >
          Connecticut
        </option>
        <option
          value="DE"
        >
          Delaware
        </option>
        <option
          value="DC"
        >
          District Of Columbia
        </option>
        <option
          value="FL"
        >
          Florida
        </option>
        <option
          value="GA"
        >
          Georgia
        </option>
        <option
          value="HI"
        >
          Hawaii
        </option>
        <option
          value="ID"
        >
          Idaho
        </option>
        <option
          value="IL"
        >
          Illinois
        </option>
        <option
          value="IN"
        >
          Indiana
        </option>
        <option
          value="IA"
        >
          Iowa
        </option>
        <option
          value="KS"
        >
          Kansas
        </option>
        <option
          value="KY"
        >
          Kentucky
        </option>
        <option
          value="LA"
        >
          Louisiana
        </option>
        <option
          value="ME"
        >
          Maine
        </option>
        <option
          value="MD"
        >
          Maryland
        </option>
        <option
          value="MA"
        >
          Massachusetts
        </option>
        <option
          value="MI"
        >
          Michigan
        </option>
        <option
          value="MN"
        >
          Minnesota
        </option>
        <option
          value="MS"
        >
          Mississippi
        </option>
        <option
          value="MO"
        >
          Missouri
        </option>
        <option
          value="MT"
        >
          Montana
        </option>
        <option
          value="NE"
        >
          Nebraska
        </option>
        <option
          value="NV"
        >
          Nevada
        </option>
        <option
          value="NH"
        >
          New Hampshire
        </option>
        <option
          value="NJ"
        >
          New Jersey
        </option>
        <option
          value="NM"
        >
          New Mexico
        </option>
        <option
          value="NY"
        >
          New York
        </option>
        <option
          value="NC"
        >
          North Carolina
        </option>
        <option
          value="ND"
        >
          North Dakota
        </option>
        <option
          value="OH"
        >
          Ohio
        </option>
        <option
          value="OK"
        >
          Oklahoma
        </option>
        <option
          value="OR"
        >
          Oregon
        </option>
        <option
          value="PA"
        >
          Pennsylvania
        </option>
        <option
          value="RI"
        >
          Rhode Island
        </option>
        <option
          value="SC"
        >
          South Carolina
        </option>
        <option
          value="SD"
        >
          South Dakota
        </option>
        <option
          value="TN"
        >
          Tennessee
        </option>
        <option
          value="TX"
        >
          Texas
        </option>
        <option
          value="UT"
        >
          Utah
        </option>
        <option
          value="VT"
        >
          Vermont
        </option>
        <option
          value="VA"
        >
          Virginia
        </option>
        <option
          value="WA"
        >
          Washington
        </option>
        <option
          value="WV"
        >
          West Virginia
        </option>
        <option
          value="WI"
        >
          Wisconsin
        </option>
        <option
          value="WY"
        >
          Wyoming
        </option>
      </select>
    </div>
    <div
      className="col-6"
    >
      <label
        htmlFor="businessZipInput"
      >
        Zip
      </label>
      <input
        className="form-control"
        id="businessZipInput"
        type="text"
      />
    </div>
  </div>
</div>
`;
